src/main/scala/org/apache/spark/deploy/armada/submit/ConfigGenerator.scala

```diff
@@ -43,8 +44,15 @@ private[submit] class ConfigGenerator(val prefix: String, val conf: SparkConf) {
 
   // Store the config files as annotations in the armada submit request
   def getAnnotations: Map[String, String] = {
-    confFiles.map(f =>
-      (prefix + "/" + f.getName, Source.fromFile(f.toString).mkString)).toMap
```

Comment:
`Using` makes sure the closable object returned by `Source.fromFile(f.toString)` is always being closed (`try { } finally { }`).

It returns a `Try` object, which is a class construct for `try { } catch { }`, providing programmatic access to the result or exception. With `Try.get`, the value is returned, or the exception is thrown.

Reply:
Thanks for the tip about Using()!

Reply:
I removed `Using`, because it adds dependency to `scala-collection-compat`, has a version conflict with Spark 3.3.4. This keeps the `createImage.sh` setup simple.